File: tombay33_ui/src/services/ItemCategoryService.js
import http from "../http-common";

class ItemCategoryService {
  getAllItemCategorys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/itemCategory/itemCategorys`, searchDTO);
  }

  get(itemCategoryId) {
    return this.getRequest(`/itemCategory/${itemCategoryId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/itemCategory?field=${matchData}`, null);
  }

  addItemCategory(data) {
    return http.post("/itemCategory/addItemCategory", data);
  }

  update(data) {
  	return http.post("/itemCategory/updateItemCategory", data);
  }
  
  uploadImage(data,itemCategoryId) {
  	return http.postForm("/itemCategory/uploadImage/"+itemCategoryId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ItemCategoryService();

File: src/main/java/com/tombay33/service/ItemCategoryService.java
package com.tombay33.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.tombay33.domain.ItemCategory;
import com.tombay33.dto.ItemCategoryDTO;
import com.tombay33.dto.ItemCategorySearchDTO;
import com.tombay33.dto.ItemCategoryPageDTO;
import com.tombay33.dto.ItemCategoryConvertCriteriaDTO;
import com.tombay33.service.GenericService;
import com.tombay33.dto.common.RequestDTO;
import com.tombay33.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ItemCategoryService extends GenericService<ItemCategory, Integer> {

	List<ItemCategory> findAll();

	ResultDTO addItemCategory(ItemCategoryDTO itemCategoryDTO, RequestDTO requestDTO);

	ResultDTO updateItemCategory(ItemCategoryDTO itemCategoryDTO, RequestDTO requestDTO);

    Page<ItemCategory> getAllItemCategorys(Pageable pageable);

    Page<ItemCategory> getAllItemCategorys(Specification<ItemCategory> spec, Pageable pageable);

	ResponseEntity<ItemCategoryPageDTO> getItemCategorys(ItemCategorySearchDTO itemCategorySearchDTO);
	
	List<ItemCategoryDTO> convertItemCategorysToItemCategoryDTOs(List<ItemCategory> itemCategorys, ItemCategoryConvertCriteriaDTO convertCriteria);

	ItemCategoryDTO getItemCategoryDTOById(Integer itemCategoryId);







}





